### ISSUE TITLE
components outside of Router unexpectedly getting destroyed

### ISSUE THREAD

**Issue by matryer** [EmilTholin/svelte-routing#103]
We have noticed that components outside the `<Router>` are getting destroyed (`onDestroy` callbacks are firing) unexpectedly, even though they should remain active (and do remain visible).

To show what I mean, I added logging to the example in this repo: https://github.com/matryer/svelte-routing/commit/5062dda39289bba85246c05e5841c0be71cc8712

VIDEO showing issue: https://www.dropbox.com/s/yr7ud91wvzbv348/RecordingOfDestroyIssue.mp4?dl=1 

We expect that the `App: onDestroy` would **not** be called since the routing hasn't moved away from that component.

In our case, we have a sidebar that lives outside a Router, that we wish to remain active while the user is navigating around the app.

**Comment by EmilTholin** [comment]
Hi @matryer!

Thank you for the reproduction and the video describing your issue. Might it be that you are declaring the `onDestroy` inside of the `onMount`? I don't get this behaviour if I declare the `onDestoy` in the top-level scope of the component script.

```javascript
  import { onMount, onDestroy } from "svelte";
  // ...

  onMount(() => {
    console.info("App: onMount");
  });

  onDestroy(() => {
    console.info("App: onDestroy");
  });
```

Instead of using `onDestroy` you could also return a function from `onMount` that will be invoked when the component is destroyed.

```javascript
  import { onMount } from "svelte";
  // ...

  onMount(() => {
    console.info("App: onMount");

    return () => {
      console.info("App: onDestroy");
    }
  });
```

**Comment by matryer** [comment]
Thanks @EmilTholin. 

Actually, we were calling `onDestroy` after a few data operations, so it was getting queued up fairly late into the lifecycle of the component. When we move it _into_ the `onMount` method, everything works as expected (in our case).

So yes, something to do with where `onDestroy` is called seems to matter.

Closing since we've resolved it, and hopefully this issue stands as a clue for future generations.
